WordPress/Classes/Networking/PlansRemote.swift

```diff
@@ -54,9 +56,173 @@ private func mapPlansResponse(response: AnyObject) throws -> (activePlan: Plan,
             return (result.0, plans)
         }
     })
+    
     guard let activePlan = parsedResponse.0 else {
         throw PlansRemote.Error.NoActivePlan
     }
     let availablePlans = parsedResponse.1.sort()
     return (activePlan, availablePlans)
 }
+
+private func parseFeatureGroups(json: [[String: AnyObject]], forPlan plan: Plan) {
+    let groups: [PlanFeatureGroup] = json.flatMap { groupJson in
+        guard let slugs = groupJson["items"] as? [String] else { return nil }
+        return PlanFeatureGroup(title: groupJson["title"] as? String, slugs: slugs)
+    }
+    
+    PlanFeatureGroup.setGroups(groups, forPlan: plan)
+}
+
+class PlanFeaturesRemote: ServiceRemoteREST {
+    
+    enum Error: ErrorType {
+        case DecodeError
+    }
+    
+    private var cacheDate: NSDate?
+    private let cacheFilename = "plan-features.json"
+    
+    func getPlanFeatures(success: PlanFeatures -> Void, failure: ErrorType -> Void) {
+        // First we'll try and return plan features from memory, then check our disk cache,
+        // and finally hit the network if we don't have anything recent enough
+        if let planFeatures = inMemoryPlanFeatures {
+            success(planFeatures)
+            return
+        }
+        
+        // If we have features cached to disk, update the in-memory list and the cache date to match
+        if let (planFeatures, date) = cachedPlanFeaturesWithDate() {
+            inMemoryPlanFeatures = planFeatures
+            cacheDate = date
+            
+            success(planFeatures)
+            return
+        }
+        
+        fetchPlanFeatures({ [weak self] planFeatures in
+            self?.inMemoryPlanFeatures = planFeatures
+            self?.cacheDate = NSDate()
+            
+            success(planFeatures)
+        }, failure: failure)
+    }
+    
+    private var _planFeatures: PlanFeatures?
+    private var inMemoryPlanFeatures: PlanFeatures? {
+        get {
+            // If we have something in memory and it's less than a day old, return it.
+            if let planFeatures = _planFeatures,
+                let cacheDate = cacheDate where
+                cacheDateIsValid(cacheDate) {
+                    return planFeatures
+            }
+            
+            return nil
+        }
+        
+        set {
+            _planFeatures = newValue
+        }
+    }
+    
+    private func cachedPlanFeatures() -> PlanFeatures? {
+        guard let (planFeatures, _) = cachedPlanFeaturesWithDate() else { return nil}
+
+        return planFeatures
+    }
+    
+    /// @return An optional tuple containing a collection of cached plan features and the date when they were fetched
```

Comment:
The new documentation format is `- returns: ...`

![screen shot 2016-03-21 at 11 24 27](https://cloud.githubusercontent.com/assets/8739/13915893/95fd9530-ef57-11e5-9316-8da8a50d5c51.png)
![screen shot 2016-03-21 at 11 24 40](https://cloud.githubusercontent.com/assets/8739/13915894/96007958-ef57-11e5-88a5-eb022f56573c.png)

https://developer.apple.com/library/ios/documentation/Xcode/Reference/xcode_markup_formatting_ref/Returns.html#//apple_ref/doc/uid/TP40016497-CH25-SW1